app/src/main/kotlin/com/intive/picover/images/viewmodel/ImagesViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package com.intive.picover.images.viewmodel
+
+import android.net.Uri
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.images.repository.ImagesRepository
+import com.intive.picover.images.viewmodel.ImagesViewModel.LatestFirebaseStorageState.Error
+import com.intive.picover.images.viewmodel.ImagesViewModel.LatestFirebaseStorageState.Loading
+import com.intive.picover.images.viewmodel.ImagesViewModel.LatestFirebaseStorageState.Success
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ImagesViewModel @Inject constructor(private val repository: ImagesRepository) : ViewModel() {
+
+	private val _uiState = MutableStateFlow<LatestFirebaseStorageState>(Loading)
+	val uiState: StateFlow<LatestFirebaseStorageState> = _uiState
+
+	init {
+		viewModelScope.launch {
+			fetchPics().collect()
+		}
+	}
+
+	private suspend fun fetchPics(): Flow<List<Uri>> = callbackFlow {
+		_uiState.value = Loading
+		runCatching {
+			val uris = repository.fetchItems().map { storageReference ->
+				repository.downloadUri(storageReference)
+			}
+			_uiState.value = Success(uris.sorted())
+			trySend(uris)
```

Comment:
in `runCatching` you should rather do only risky operations and then in `onSuccess` / `onFailure` react to the result